src/services/ReactService.ts

```diff
@@ -0,0 +1,48 @@
+
+import * as Discord from 'discord.js';
+
+import {AutoWired, Inject, Singleton} from 'typescript-ioc';
+import { BaseService } from '../base/BaseService';
+import { ErrorMessageService } from './ErrorMessageService';
+
+@Singleton
+@AutoWired
+export class ReactService extends BaseService {
+
+  @Inject private errorMessageService: ErrorMessageService;
+
+  public async init(client) {
+    await super.init(client);
+  }
+
+  /**
+   * Temporarily react to a target message with a specific emoji.
+   *
+   * @param targetMessage The message to react to.
+   * @param emoji The emoji to use for the reaction.
+   * @param userCommandMessage The user command that instructed to apply the emoji.
+   */
+  async tempReactToMessage(targetMessage: Discord.Message, emoji: Discord.Emoji,
+                           userCommandMessage: Discord.Message): Promise<boolean> {
+    try {
+      let reaction = await targetMessage.react(emoji);
+      if (userCommandMessage.deletable) {
+        await userCommandMessage.delete();
+      }
+
+      setTimeout(async () => {
+        await reaction.remove();
+      }, 20000);

```

Comment:
Maybe this should be customizable? Pass in a duration = 20000 to the function so it defaults to 20000 but the implementer can choose. Or, it should be made a constant somewhere.

Reply:
- [ ] Pass in a duration = 20000 to the function so it defaults to 20000 but the implementer can choose